src/sentry/demo/data_population.py

```diff
@@ -1081,55 +1045,71 @@ def populate_generic_transaction(
 
         self.log_info("populate_generic_error.finished")
 
-    def populate_sessions(self, project, error_file):
+    @catch_and_log_errors
+    def populate_sessions(self, project, distribution_fn_num: int, mobile: bool):
         self.log_info("populate_sessions.start")
+
         dsn = ProjectKey.objects.get(project=project)
 
-        react_error = get_event_from_file(error_file)
+        # keep track of versions for mobile
+        seen_versions = []
+        num_versions = 0
 
-        for (timestamp, day) in self.iter_timestamps(4):
-            transaction_user = self.generate_user()
-            sid = uuid4().hex
+        for (timestamp, day) in self.iter_timestamps(distribution_fn_num):
             release = get_release_from_time(project.organization_id, timestamp)
             version = release.version
 
-            # initialize the session
-            session_data = {
-                "init": True,
-            }
-            self.send_session(sid, transaction_user["id"], dsn, timestamp, version, **session_data)
+            # add new version if necessary
+            if version not in seen_versions:
+                seen_versions.append(version)
+                num_versions += 1
 
-            # determine if this session should crash or exit with success
-            rate_by_release_num = (
-                crash_free_rate_by_release_quick if self.quick else crash_free_rate_by_release
-            )
-            # get the release num from the last part of the version
-            release_num = int(version.split(".")[-1])
-            threshold = rate_by_release_num[release_num]
-            outcome = random.random()
-            if outcome > threshold:
-                # if crash, make an error for it
-                local_event = copy.deepcopy(react_error)
-                local_event.update(
-                    project=project,
-                    platform=project.platform,
-                    timestamp=timestamp,
-                    user=transaction_user,
-                    release=version,
-                )
-                update_context(local_event)
-                self.fix_error_event(local_event)
-                self.safe_send_event(local_event)
+            formatted_time = timestamp.isoformat()
+            envelope_headers = "{}"
+            item_headers = json.dumps({"type": "sessions"})
 
-                data = {
-                    "status": "crashed",
-                }
+            agg = []
+
+            if self.quick:
+                num_users = int(random.uniform(200, 400))
             else:
-                data = {
-                    "status": "exited",
+                num_users = int(random.uniform(1000, 2000))
+
+            # create session data for each user
+            for _ in range(num_users):
+                exited = random.choices([1, 2, 3, 4], k=1, weights=[10, 5, 3, 1])[0]
+                rand = random.random()
+                if rand <= 0.0045:
```

Comment:
@roggenkemper we should avoid magic numbers like this. Make it a variable somehwere.

Reply:
makes sense